controllers/apps/transformer_component_service.go

```diff
@@ -267,21 +270,70 @@ func (t *componentServiceTransformer) createOrUpdateService(ctx graph.TransformC
 	}
 
 	if podService && kind == multiClusterServicePlacementInUnique {
-		return t.createOrUpdateServiceInUnique(ctx, dag, graphCli, service, owner)
+		// create or update service in unique, by hacking the pod placement strategy.
+		ordinal := func() int {
+			subs := strings.Split(service.GetName(), "-")
+			o, _ := strconv.Atoi(subs[len(subs)-1])
+			return o
+		}
+		multicluster.Assign(ctx.GetContext(), service, ordinal)
+	}
+
+	createOrUpdateService := func(service *corev1.Service) error {
+		key := types.NamespacedName{
+			Namespace: service.Namespace,
+			Name:      service.Name,
+		}
+		obj := &corev1.Service{}
+		if err := ctx.GetClient().Get(ctx.GetContext(), key, obj, inDataContext4C()); err != nil {
+			if apierrors.IsNotFound(err) {
+				graphCli.Create(dag, service, inDataContext4G())
+				return nil
+			}
+			return err
+		}
+
+		// don't update service not owned by the owner, to keep compatible with existed cluster
+		if !model.IsOwnerOf(owner, obj) {
+			return nil
+		}
+
+		objCopy := obj.DeepCopy()
+		objCopy.Spec = service.Spec
+
+		// if skip immutable check, update the service directly
+		if skipImmutableCheckForComponentService(obj) {
+			resolveServiceDefaultFields(&obj.Spec, &objCopy.Spec)
+			if !reflect.DeepEqual(obj, objCopy) {
+				graphCli.Update(dag, obj, objCopy, inDataContext4G())
+			}
+			return nil
+		}
+		// otherwise only support to update the params associated with appsv1alpha1.ClusterComponentService and further change
```

Comment:
-> otherwise only support to update the override params defined in cluster.spec.componentSpec[].services